src/core/util/tag.h

```diff
@@ -56,8 +56,10 @@ class Tag
 
   Tag() : m_Buf{} {}
 
-  Tag(const std::uint8_t* buf)
+  Tag(const std::uint8_t* buf) : m_Buf{}
   {
+    if (!buf)
```

Comment:
Might as well add an assertion before this check.

Reply:
Should the assertion replace the check, or should I use both?

Reply:
Both because we won't always define NDEBUG.